components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -110,10 +111,13 @@ base::Value::List BuildMessages(
 }  // namespace
 
 EngineConsumerOAIRemote::EngineConsumerOAIRemote(
-    const mojom::CustomModelOptions& model_options,
+    const mojom::Model& model,
     scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory) {
-  model_options_ = model_options;
-  max_page_content_length_ = kCustomModelMaxPageContentLength;
+  model_options_ = *model.options->get_custom_model_options();
+  max_associated_content_length_ =
+      ModelService::GetMaxAssociatedContentLengthForModel(model);
```

Comment:
Isn't this a circular reference since `ModelService` constructs `EngineConsumerOAIRemote`? Maybe it's ok but either way I don't think there's an #include for `ModelService` in this file. Perhaps it could be in a utils file? Just a nit anyway...

Reply:
Now that we're calculating the MACL earlier, we can retrieve it directly from the options. Good catch!